Files/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -824,14 +827,17 @@ public async Task<ReturnResult> MoveItemsFromClipboard(DataPackageView packageVi
             ReturnResult returnStatus = ReturnResult.InProgress;
 
             var destinations = new List<string>();
-            List<ShellFileItem> binItems = null;
             foreach (var item in source)
             {
                 if (recycleBinHelpers.IsPathUnderRecycleBin(item.Path))
                 {
-                    binItems ??= await recycleBinHelpers.EnumerateRecycleBin();
-                    var matchingItem = binItems.FirstOrDefault(x => x.RecyclePath == item.Path); // Get original file name
-                    destinations.Add(PathNormalization.Combine(destination, matchingItem?.FileName ?? item.Name));
+                    List<ShellFileItem> binItems = await recycleBinHelpers.EnumerateRecycleBin();
```

Comment:
I agree with the null check below but `EnumerateRecycleBin` really shouldn't be put inside the loop.

Reply:
Thanks! A little oversight from me